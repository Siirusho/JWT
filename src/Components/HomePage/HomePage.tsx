import { Grid, makeStyles, Theme, createStyles } from "@material-ui/core";
import NavBar from "../NavBar/NavBar";
import LeftBar from "../LeftBar/LeftBar";
import UsersList from "../UsersList/UsersList";
import RightBar from "../RightBar/RightBar";
import { FC } from 'react';

const useStyles = makeStyles<Theme>(theme =>
    createStyles({
        right: {
            [theme.breakpoints.down("sm")]: {
                display: "none",
            }
        }
    }));

const HomePage: FC = () => {
    const classes = useStyles();
    return (
        <div className="App">
            <NavBar />
            <Grid container>
                <Grid item sm={2} xs={2}>
                    <LeftBar />
                </Grid>
                <Grid item sm={7} xs={10}>
                    <UsersList />
                </Grid>
                <Grid item sm={3} className={classes.right}>
                    <RightBar />
                </Grid>
            </Grid>
        </div>
    );
};

export default HomePage;