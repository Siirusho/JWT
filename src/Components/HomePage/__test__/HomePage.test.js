import React from "react";
import HomePage from "../HomePage";
import { render } from "@testing-library/react";
import listReducer, {listState} from "../../../store/List/_reducer";
import { createStore } from "redux";
import { Provider } from "react-redux";

function renderWithProviders(ui, { reduxState } = {}) {
    const store = createStore(listReducer, reduxState || listState);
    return render(<Provider store={store}>{ui}</Provider>);
};


describe("Home page", () => {
    it("should render correct HomePage component and match with snapshot", () => {
        const { asFragment } = renderWithProviders(<HomePage />);
        expect(asFragment(<HomePage />)).toMatchSnapshot();
    })
});