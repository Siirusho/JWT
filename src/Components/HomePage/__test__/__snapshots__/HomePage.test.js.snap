// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correct HomePage component and match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navBar-13 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-5 MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-logoLg-3 MuiTypography-h6"
          style="cursor: pointer;"
        >
          Home Page
        </h6>
        <h6
          class="MuiTypography-root makeStyles-logoSm-4 MuiTypography-h6"
        >
          Home
        </h6>
        <div
          class="makeStyles-disable-9 makeStyles-search-6"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-searchIcon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiInputBase-root makeStyles-input-10"
          >
            <input
              class="MuiInputBase-input"
              placeholder="Search..."
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="makeStyles-badgeContainer-11"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-searchButton-2"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiBadge-root makeStyles-badge-12"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangular MuiBadge-colorSecondary"
            >
              4
            </span>
          </span>
          <span
            class="MuiBadge-root makeStyles-badge-12"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangular MuiBadge-colorSecondary"
            >
              2
            </span>
          </span>
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2"
      >
        <div
          class="MuiContainer-root makeStyles-container-17 MuiContainer-maxWidthLg"
        >
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              HomePage
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Friends
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              List
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Music
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Videos
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h18V4H4c-1.1 0-2 .9-2 2v11H0v3h14v-3H4V6zm19 2h-6c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 9h-4v-7h4v7z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Apps
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Collections
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Settings
            </p>
          </div>
          <div
            class="makeStyles-item-14"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-15"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <p
              class="MuiTypography-root makeStyles-text-16 MuiTypography-body1"
            >
              Log Out
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7"
      >
        <div
          class="makeStyles-loaderWrapper-21"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 400px; height: 400px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="21"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-right-1 MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          class="MuiContainer-root makeStyles-container-19 MuiContainer-maxWidthLg"
        >
          <p
            class="MuiTypography-root makeStyles-title-20 MuiTypography-body1"
          >
            Online Friends
          </p>
          <div
            class="MuiAvatarGroup-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar"
              style="z-index: 5;"
            >
              <img
                alt="Remy Sharp"
                class="MuiAvatar-img"
                src="https://material-ui.com/static/images/avatar/1.jpg"
              />
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar"
              style="z-index: 4;"
            >
              <img
                alt="Travis Howard"
                class="MuiAvatar-img"
                src="https://material-ui.com/static/images/avatar/2.jpg"
              />
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar"
              style="z-index: 3;"
            >
              <img
                alt="Cindy Baker"
                class="MuiAvatar-img"
                src="https://material-ui.com/static/images/avatar/3.jpg"
              />
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar"
              style="z-index: 2;"
            >
              <img
                alt="Agnes Walker"
                class="MuiAvatar-img"
                src="https://material-ui.com/static/images/avatar/4.jpg"
              />
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar"
              style="z-index: 1;"
            >
              <img
                alt="Trevor Henderson"
                class="MuiAvatar-img"
                src="https://material-ui.com/static/images/avatar/5.jpg"
              />
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Gallery
          </p>
          <ul
            class="MuiImageList-root"
            style="margin: -2px;"
            sx="[object Object]"
          >
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://images.unsplash.com/photo-1516802273409-68526ee1bdd6?w=164&h=164&fit=crop&auto=format"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://images.unsplash.com/photo-1522770179533-24471fcdba45?w=164&h=164&fit=crop&auto=format"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://images.unsplash.com/photo-1533827432537-70133748f5c8?w=164&h=164&fit=crop&auto=format"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://images.unsplash.com/photo-1444418776041-9c7e33cc5a9c?w=164&h=164&fit=crop&auto=format"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://images.unsplash.com/photo-1551963831-b3b1ca40c98e?w=164&h=164&fit=crop&auto=format"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://wallpaper-mania.com/wp-content/uploads/2018/09/High_resolution_wallpaper_background_ID_77701312990.jpg"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="http://images6.fanpop.com/image/photos/41400000/amazing-sunset-photographs-greyswan618-41405529-1600-1000.jpg"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://i.pinimg.com/originals/73/a4/51/73a4514cb61dbe836b403a50bb63cc70.jpg"
                />
              </div>
            </li>
            <li
              class="MuiImageListItem-root"
              style="width: 50%; height: 104px; padding: 2px;"
            >
              <div
                class="MuiImageListItem-item"
              >
                <img
                  src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQIzS0IRO5I_tlDbIBb0k0_ydRL6FdDNbFu9WX7kr-7GE3UJ2XDugo1TpG-25xq22OuPVw&usqp=CAU"
                />
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
